### DISCUSSION TITLE
Whats going on here, Cron expression didn't work out

### DISCUSSION THREAD

**Discussion by GerhardKreuzer** [quartznet/quartznet#767] [General] [answered]
Hi,
I am new to cron expressions, so I tried this:
trigger = TriggerBuilder.Create() _
            .WithIdentity("triggerConsultationCheck", "BackOffice") _
            .WithDescription("daily") _
            .WithCronSchedule("0 0 3 * * MON-FRI") _
            .Build()
And I am pretty sure, that this Trigger fires on all days (Momday to friday) as 3.00 in the very early morning.
But I get a Format exception, telling me, that it is not supported to have a day-of-month AND day-of-week specifier. Ok, thats clear, and thats the reason that I wrote an '*' ....

Whats wrong here?

Thanks for helping.

With best regards

Gerhard

**Comment by GerhardKreuzer** [comment]
Addition:
I get my 'knowedge' from Lesson 6, Cron Triggers:
Cron Expressions

Cron-Expressions are used to configure instances of CronTrigger. Cron-Expressions are strings that are actually made up of seven sub-expressions, that describe individual details of the schedule. These sub-expression are separated with white-space, and represent:
• 1.Seconds

• 1.Minutes

• 1.Hours

• 1.Day-of-Month

• 1.Month

• 1.Day-of-Week

• 1.Year (optional field)


An example of a complete cron-expression is the string “0 0 12 ? * WED” - which means “every Wednesday at 12:00 pm”.

Individual sub-expressions can contain ranges and/or lists. For example, the day of week field in the previous (which reads “WED”) example could be replaces with “MON-FRI”, “MON, WED, FRI”, or even “MON-WED,SAT”.

Wild-cards (the ‘’ character) can be used to say “every” possible value of this field. Therefore the ‘’ character in the “Month” field of the previous example simply means “every month”. A ‘*’ in the Day-Of-Week field would obviously mean “every day of the week”.

But maybe this Infos were outdated.

With best regards

Gerhard

**Comment by GerhardKreuzer** [comment]
Addendum:
WithCronSchedule("0 0 3 * * ?") .. works
WithCronSchedule("0 0 3 * * MON") .. Fails

Error: not supported to have a day-of-month AND day-of-week specifier

Ok, I am pretty sure, I only have a day-of-week specified ... 

With best regards

Gerhard

**Comment by GerhardKreuzer** [comment]
Addendum:
WithCronSchedule("0 0 3 * * *") .. Fails too

**Comment by GerhardKreuzer** [comment]
Addendum:
WithCronSchedule("0 0 3 ? * MON-FRI") .. works, this means I dont get an runtime error and will see tomorrow, whats really going on.

Did anybody have an explaination for that behaviour?

Thanks a lot

With best regards

**Comment by lahma** [comment, marked as answer]
Please see http://www.cronmaker.com/ for an easy to use tool for generating expressions. Think '?' as ignore flag here.

> The ‘?’ character is allowed for the day-of-month and day-of-week fields. It is used to specify “no specific value”. This is useful when you need to specify something in one of the two fields, but not the other.
